src/loaders.ts

```diff
@@ -16,6 +17,18 @@ const loadJs: LoaderSync = function loadJs(filepath) {
   return result;
 };
 
+const loadJs: LoaderAsync = async function loadJs(filepath) {
+  // This logic is validated by tests running in Node versions that don't
+  // support dynamic imports, like Node 10.
+  /* istanbul ignore else */
+  if (canUseDynamicImport()) {
+    const imported = await import(filepath);
```

Comment:
An import call will syntax error in node versions that do not support it. In a [PR](https://github.com/substack/tape/pull/547) I'm fiddling with to support ESM in [tape](https://github.com/substack/tape) I side step this syntax error by conditionally requiring another file containing the import statement...

The conditional require:
https://github.com/lohfu/tape/blob/esm-support/bin/tape#L29

The separate file:
https://github.com/lohfu/tape/blob/esm-support/bin/load-module-esm-support.js

Since dynamic imports seem to be up to 5 times slower than require, we conditionally use import or require by checking file extensions and [get-package-type](https://github.com/cfware/get-package-type). As you probably load far less files this might not be necessary for you.

Reply:
If you dont want to conditionally use require vs import, you could just use the function you create to check dynamic import support, https://github.com/davidtheclark/cosmiconfig/pull/251/files#diff-e61e569af4ed5c2bd19c648e11225dd502296f0931b339c88cb82288a493e6adR6.

ie something like:

```ts
let dynamicImport;

try {
  dynamicImport = new Function('id', 'return import(id);');
} catch (e) {} 

// later

const loadJs: LoaderAsync = async function loadJs(filepath) {
  if (dynamicImport()) {
    const imported = await dynamicImport(filepath);

    return imported.default;
  } else {
    return loadJsSync(filepath, null);
  }
};
```

Reply:
it appears it's even trickier... versions 10.0.0 - 12.16 and 13.0 - 13.1 support the `import` keyword, but will throw a "Not Supported" error when called. See https://github.com/inspect-js/has-dynamic-import/issues/2 for an attempted work around.